### ISSUE TITLE
librsync-2.0.0 fails Signature/md4 test on big-endian platforms (powerpc)

### ISSUE THREAD

**Issue by trofi** [librsync/librsync#132]
Originally reported here: https://bugs.gentoo.org/635834

The tests fail on powerpc32 (powerpc-unknown-linux-gnu) and powerpc64 (powerpc64-unknown-linux-gnu) linux as:
```
3/7 Test #3: Signature ........................***Failed    0.34 sec
```

The cause appears to be simple: md4 has explicit code to handle WORDS_BIGENDIAN platforms but config.h does not set it correctly on anything except something-Apple:

src/config.h.cmake:

```c
  /* Define WORDS_BIGENDIAN to 1 if your processor stores words with the most
     significant byte first (like Motorola and SPARC, unlike Intel). */
  #if defined AC_APPLE_UNIVERSAL_BUILD
  # if defined __BIG_ENDIAN__
  #  define WORDS_BIGENDIAN 1
  # endif
  #else
  # ifndef WORDS_BIGENDIAN
  /* #  undef WORDS_BIGENDIAN */
  # endif
  #endif
```

Changing it to:

```c
  # if defined __BIG_ENDIAN__
  #  define WORDS_BIGENDIAN 1
  # endif
```
makes tests pass on powerpc32/64.

gcc defines a few macros for it:
```c
$ powerpc64-unknown-linux-gnu-gcc -dM -E - </dev/null | grep -i endian
#define __BIG_ENDIAN__ 1
#define _BIG_ENDIAN 1
```

**Comment by DerDakon** [comment]
cmake --help-module TestBigEndian

**Comment by dbaarda** [comment]
Can you please check this with v2.0.1 as I'm pretty sure it was fixed in #113.

**Comment by trofi** [comment]
2.0.1 (and master branch) works fine:
```
      Start  8: Signature
 8/12 Test  #8: Signature ........................   Passed    0.24 sec
```
Thank you!

My apologies! I should have checked before reporting the bug.
